benchmarks/benchmarks/esmf_regridder/__init__.py

```diff
@@ -452,3 +452,37 @@ def time_save(self, _, tp, rgt):
     def time_load(self, _, tp, rgt):
         """Benchmark the loading time."""
         _ = self.load_regridder(self.source_file)
+
+
+class TimeMaskedRegridding:
+    """Benchmarks for :class:`~esmf_regrid.esmf_regrid.schemes.ESMFAreaWeightedRegridder`."""
+
+    def setup(self):
+        """ASV setup method."""
+        src = _curvilinear_cube(250, 251, [-180, 180], [-90, 90])
+        tgt = _curvilinear_cube(251, 250, [-180, 180], [-90, 90])
+
+        # Make src and tgt discontiguous at (0, 0)
+        src_mask = np.zeros([250, 251], dtype=bool)
+        src_mask[0, :] = True
+        src.data = np.ma.array(src.data, mask=src_mask)
+        src.coord("latitude").bounds[0, :, :2] = 0
+        src.coord("longitude").bounds[0, :, :2] = 0
+
+        tgt_mask = np.zeros([251, 250], dtype=bool)
+        tgt_mask[:, 0] = True
+        tgt.data = np.ma.array(tgt.data, mask=tgt_mask)
+        tgt.coord("latitude").bounds[:, 0, ::3] = 0
+        tgt.coord("longitude").bounds[:, 0, ::3] = 0
+
+        self.regrid_class = ESMFAreaWeightedRegridder
+        self.src = src
+        self.tgt = tgt
+
+    def time_prepare_without_masks(self):
+        """Benchmark the prepare time with discontiguities but no mask."""
+        _ = self.regrid_class(self.src, self.tgt)
+
+    def time_prepare_with_masks(self):
```

Comment:
Please could both these be put into a single benchmark, using ASV parameters for with/without masks. This is appropriate for anything that is designed to be compared, and ASV is designed to make this easy when plotting data.

Here is an existing example of using parameterisation:

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/55b8354dfe8aa7e2d3e9dd33f76647cbbb0b5e76/benchmarks/benchmarks/esmf_regridder/__init__.py#L59-L71

Reply:
Yes I see your point. Not looking for a permanent comparison - this is an improvement and as such we can see the improvement by comparing between commits (https://github.com/SciTools-incubator/iris-esmf-regrid/pull/219#discussion_r1144713802).